packages/react-native/src/primitives/PasswordField/__tests__/PasswordField.spec.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react-native';
+import PasswordField from '../PasswordField';
+import { icons } from '../../../assets';
+
+const labelText = 'Label';
+const testID = 'passwordInput';
+const defaultProps = {
+  testID: testID,
+  label: labelText,
+};
+
+describe('PasswordField', () => {
+  it('renders as expected', async () => {
+    const { toJSON, findAllByRole, getByTestId, getByText, getByRole } = render(
+      <PasswordField {...defaultProps} />
+    );
+    expect(toJSON()).toMatchSnapshot();
+
+    expect(await findAllByRole('text')).toHaveLength(1);
+    const textInput = getByTestId(testID);
+    expect(textInput.props.editable).toBe(true);
+    expect(textInput.props.secureTextEntry).toBe(true);
+    expect(textInput.props.accessible).toBe(true);
+    const image = getByRole('image');
+    expect(image.props.source).toBe(icons.visibilityOff);
+    const label = getByText(labelText);
+    expect(label).not.toBeNull();
+  });
+
+  it('toggles visibility', () => {
+    const { getByRole } = render(<PasswordField {...defaultProps} />);
+
+    const icon = getByRole('image');
+    expect(icon.props.source).toBe(icons.visibilityOff);
+    fireEvent.press(icon);
+    expect(icon.props.source).toBe(icons.visibility);
+  });
```

Comment:
Is there also a way to check if the password is no longer visible? As in, when visibility is off, it should display dots instead of text, correct?

Reply:
I added a couple assertions about `secureTextEntry` but I think testing the input value goes a bit deep into the functionality of the react-native input component which we shouldn't be testing imo.